src/main/java/view/RecSongView.java

```diff
@@ -27,7 +27,7 @@ public RecSongView() {
 
         builder.addLabel("New Song: placeholder_name");
 
-        builder.addButton("Add to Playlist");
+        builder.addButton("addToPlaylist", "Add to Playlist");
 
         view = builder.build();
```

Comment:
instead, it is more common to write view = builder.addLabel("New Song: placeholder_name")
                                                                                 .addButton("addToPlaylist", "Add to Playlist")
                                                                                 .setController(recSongController)
                                                                                 .build()

Reply:
Also, I think that it would be better to have a Controller class (with all other types of Controllers being subclasses) so that in our ViewBuilder we can have a setController() method that will simplify things. Side note: we can talk about this in our next meeting. But I can implement this myself.

Reply:
Nvm, the second comment about setController() method is absolute garbage, don't mind me at all.

Reply:
I also figured out the constructor thing by myself, so no need to do anything on your own. Plz approve my pull request though